composer/trainer/trainer.py

```diff
@@ -232,6 +234,16 @@ def _is_cuda_oom(e: RuntimeError):
     return False
 
 
+def _fsdp_reshard_and_cleanup(model: torch.nn.Module):
+    for name, module in model.named_modules():
+        if isinstance(module, FullyShardedDataParallel):
+            if module.check_is_root():
```

Comment:
Can you add a comment here explaining why to only apply this on root? Since its using a private API :)